app/services/shorten_url.rb

```diff
@@ -0,0 +1,20 @@
+class ShortenUrl
+
+  SHORTEN_URL_LOG = 'log/tinyurl.log'
+  SUCCESS_CODES = [200, 201, 202].freeze
+
+  def self.short(url)
+    response = HTTParty.get("http://tinyurl.com/api-create.php?url=#{url}")
+    raise HTTParty::Error if response.match?(/error/i) || !SUCCESS_CODES.include?(response.code)
+    response
+  rescue HTTParty::Error => error
+    ActivityLogger.open(SHORTEN_URL_LOG, 'TINYURL_API', 'shortening url', false) do |log|
+      log.record('error', "Exception: #{error.message}")
+      log.record('error', "Attempted URL: #{url}")
```

Comment:
One more change requested (I missed this before) - we should handle the situation that occurs if `response` is nil. 

This is because `response` will be `nil` if HTTParty raises the exception.  If _we_ raise it, then it will not be `nil`. 

Something like:
```
log.record('error', "Exception: #{error.message}")
if response
  log.record('error', "Attempted URL: #{url}")
  log.record('error', "Response body: #{response.body}")
  log.record('error', "HTTP code: #{response.code}")
else
  log.record('error', 'Exception raised by HTTParty')
end
```